### ISSUE TITLE
MzTab-M: RegEx for Adduct and question

### ISSUE THREAD

**Issue by timosachsenberg** [HUPO-PSI/mzTab#62]
Hi,
I put together a RegEx for the way could encode adducts and several possibilities arise:
https://regex101.com/r/9gcJZG/2

if we want to encode: [kM+nAdduct]charge(+|-)
e.g.:
[4M+2NH4]4+
[M-H]1-
we could use:
```
\[\d*M(\+|\-)\d*(([A-Z]\d*)+)\]\d+(\+|\-)
```
if we also want to allow:
[M-H]-
we could use:
```
\[\d*M(\+|\-)\d*(([A-Z]\d*)+)\]((\+|\-)|\d+(\+|\-))
```

Open question: How to encode multiple adducts on the same molecule: [M+Na+CH3OH]+ ?
The spec could be clearer (e.g., by adding more examples)

The spec is also a bit weak on the SME section that seems to contain quite a few copy and paste artefacts from the SMF section. I think this needs to be discusses and reworked a bit.

**Comment by timosachsenberg** [comment]
also note that examples in the spec document might need to be reworked for consistency.
e.g. [M+Na] should be [M+Na]+ (or 1+) and there are some occurrences of "Nils Hoffmann4" in the text.

**Comment by nilshoffmann** [comment]
The latter is fixed by #62.

**Comment by andrewrobertjones** [comment]
@jmrein  double checking the proposal

**Comment by nilshoffmann** [comment]
Fixed in #115
